hr_infraction/models/hr_infraction.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
+# © 2016 OpenSynergy Indonesia
+# @author Andhitia Rama
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import time
+from openerp import models, fields, api
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT
+
+
+class HrInfractionCatagory(models.Model):
+    _name = 'hr.infraction.category'
+    _description = 'Infraction Type'
+    name = fields.Char(
+        string='Name',
+        required=True,
+    )
+    code = fields.Char(
+        string='Code',
+        required=True,
+    )
+
+
+class HrInfractionWarning(models.Model):
+    _name = 'hr.infraction.warning'
+    _description = 'Infraction Warning'
+
+    name = fields.Char(
+        string='Warning',
+        required=True,
+        )
+
+    code = fields.Char(
+        string='Code',
+        )
+
+    category_ids = fields.Many2many(
+        string='Infraction Category(s)',
+        comodel_name='hr.infraction.category',
+        relation='rel_inf_warn_categ',
+        column1='warning_id',
+        column2='category_id',
+        )
+
+
+class HrInfraction(models.Model):
+    _name = 'hr.infraction'
+    _description = 'Infraction'
+    _inherit = ['mail.thread']
+
+    @api.model
+    def _default_company_id(self):
+        return self.env.user.company_id.id
+
+    name = fields.Char(
+        string='# Infraction',
+        required=False,
+        readonly=True,
+        default='/',
+        copy=False,
+    )
+    date = fields.Date(
+        string='Date',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=time.strftime(DEFAULT_SERVER_DATE_FORMAT),
```

Comment:
You can use here fields.Date.today(), that will give you the date in the server format, so no need to import it...